Documentation/driver-api/ancillary_bus.rst

```diff
@@ -0,0 +1,200 @@
+
+===========
+Ancillary Bus
+===========
+
+See <linux/ancillary_bus.h> for the models for ancillary_device and
+ancillary_driver.
+
+In some subsystems, the functionality of the core device (PCI/ACPI/other) may
+be too complex for a single device to be managed as a monolithic block or
+a part of the functionality might need to be exposed to a different subsystem.
+Splitting the functionality into smaller orthogonal devices would make it
+easier to manage data, power management and domain-specific communication with
+the hardware. A key requirement for such a split is that there is no dependency
+on a physical bus, device, register accesses or regmap support. These
+individual devices split from the core cannot live on the platform bus as they
+are not physical devices that are controlled by DT/ACPI. The same argument
+applies for not using MFD in this scenario as it relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP handles multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The ancillary bus is intended to be minimal, generic and avoid domain-specific
+assumptions. Each ancillary_device represents a part of its parent
+functionality. The generic behavior can be extended and specialized as needed
+by encapsulating an ancillary_device within other domain-specific structures and
+the use of .ops callbacks. Devices on the same ancillary bus do not share any
+structures and the use of a communication channel with the parent is
+domain-specific.
+
+When Should the Ancillary Bus Be Used
+=====================================
+
+When a kernel driver registers either an ancillary_device or ancillary_driver on
+the ancillary bus, we can use the nomenclature to refer to this kernel driver as
+a registering driver.  It is the entity that will allocate memoy for and
+register something on the ancillary bus.
+
+The ancillary bus is to be used when multiple registering drivers, who share a
+common header file, need a mechanism to make connection and provide access to a
+shared struct allocated by the registering driver(s) that registered the
+ancillary_device(s).  Each of these registering drivers can be from the same
+subsystem, or from multiple subsystems.
+
+The emphasis here is on a common generic interface that keeps subsystem
+customization out of the bus infrastructure.
+
+When Should the Ancillary Bus Not Be Used
+=========================================
+
+If there is a need for a heavier weight bus that provides functionality beyond
+just the matching up of multiple registering drivers, and sacrifices common
```

Comment:
well no, it would be the matching of multiple clients. The registering driver registers ancillary devices, it *cannot* be involved in the matching part for the ancillary drivers.